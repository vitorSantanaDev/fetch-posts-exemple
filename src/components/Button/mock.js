export const buttonProps = {
  text: 'Load More Posts',
  onClick: jest.fn(),
  disabled: false,
};
