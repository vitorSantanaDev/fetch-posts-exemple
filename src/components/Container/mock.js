import React from 'react';

export const propsContainer = {
  children: <h1>Children</h1>,
};
