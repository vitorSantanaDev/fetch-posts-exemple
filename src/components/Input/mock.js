export const propsInput = {
  value: '',
  onChange: jest.fn(),
};
