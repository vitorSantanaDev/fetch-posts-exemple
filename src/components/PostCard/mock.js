export const postCardProps = {
  title: 'title one',
  body: 'Body one',
  cover: 'img/img.png',
};
