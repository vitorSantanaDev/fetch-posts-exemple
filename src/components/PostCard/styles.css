.postContent {
  padding: 32px;
}

.post {
  background-color: #fff;
  transition: transform .3s ease-in-out;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.2);
}

.post:hover {
  transform: scale(1.05);
}

.post img {
  max-width: 100%;
}