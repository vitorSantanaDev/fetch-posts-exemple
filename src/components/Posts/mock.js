export const postsProps = [
  { id: 1, title: 'Title Post one', body: 'Body text Post one', cover: '/image/image.png' },
  { id: 2, title: 'Title Post two', body: 'Body text Post two', cover: '/image/image.png' },
  { id: 3, title: 'Title Post three', body: 'Body text Post three', cover: '/image/image.png' },
];
