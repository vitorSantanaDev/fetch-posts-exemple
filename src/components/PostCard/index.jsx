import React from 'react';
import './styles.css';
import Types from 'prop-types';

const PostCard = ({ cover, title, body }) => {
  return (
    <div className="post">
      <img src={cover} alt={title} />
      <div className="postContent">
        <h2>{title}</h2>
        <p>{body}</p>
      </div>
    </div>
  );
};

PostCard.propTypes = {
  cover: Types.string.isRequired,
  title: Types.string.isRequired,
  body: Types.string.isRequired,
};

export default PostCard;
