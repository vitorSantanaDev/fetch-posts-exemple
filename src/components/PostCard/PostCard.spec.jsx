import { render, screen } from '@testing-library/react';
import PostCard from '.';
import { postCardProps } from './mock';

describe('<PostCard />', () => {
  it('should render PostCard correctly', () => {
    render(<PostCard {...postCardProps} />);

    expect(screen.getByAltText(postCardProps.title)).toHaveAttribute('src', postCardProps.cover);
    expect(screen.getByRole('heading', { name: postCardProps.title })).toBeInTheDocument();
    expect(screen.getByText(postCardProps.body)).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<PostCard {...postCardProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
