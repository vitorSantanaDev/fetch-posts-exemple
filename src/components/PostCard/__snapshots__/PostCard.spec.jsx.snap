// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="title one"
    src="img/img.png"
  />
  <div
    class="postContent"
  >
    <h2>
      title one
    </h2>
    <p>
      Body one
    </p>
  </div>
</div>
`;
